File: frontend/src/types/Mission.ts
import { Plateau, PlateauBase } from "./Plateau"
import { MarsRoverToSend } from "./MarsRover"

export interface MissionRequest {
  plateau_in: PlateauBase,
  rover_list_in: MarsRoverToSend[]
}

export interface MissionsResponse {
  plateau: Plateau
}

File: backend/app/api/models/mission.py
from typing import List
from pydantic import BaseModel
from .plateau import PlateauBase, Plateau
from .mars_rover import MarsRoverIn


class MarsMissionSetup(BaseModel):
    plateau_in: PlateauBase
    rover_list_in: List[MarsRoverIn]

class MarsMissionSchema(BaseModel):
    plateau: Plateau
